solvers/NonlinearProgram.m

```diff
@@ -155,7 +155,6 @@
       obj.bbcon_lb = [];
       obj.bbcon_ub = [];
       
-      obj = obj.setSolver('default');
```

Comment:
I see a problem that probably is related to what you were trying to tell me: that we won't be able to set "manually" the solver and will always be set to default before starting the optimization.  Ok, I'm thinking about a solution.